docs/designs/checkpoint_design.md

```diff
@@ -0,0 +1,59 @@
+# ElasticDL Training Checkpoints Design
+This document describes the design of save checkpoints for ElasticDL training.
+
+## Motivation
+Checkpoints capture the exact value of all parameters (tf.Variable objects and embedding vectors) used by a model. In ElasticDL, we need to save all parameters of the model to checkpoints during training for the following reasons:
+* We can export the model with the best evaluation metrics during training from checkpoints. Generally, we evaluate the model and monitor the evaluation metrics during training. We will stop training when we find the evaluation metrics don't go better or even worse due to over-fitting. So, the final parameters may not be the best. With checkpoints, we can choose a checkpoint with the best evaluation metrics to export for serving.
```

Comment:
Use early stop here which is widely accepted. You can add explanations after it.

Reply:
Yes!